### PR TITLE
Adding volume-wise standardization normalization for the sct_deepseg_gm

### PR DESCRIPTION
**Pull request by perone**
This PR will add:

- Two new classes for normalization: `StandardizationTransform` and `VolumeStandardizationTransform`. One is intended to use if we want to go back to the previous normalization mechanism and the other is the new one for doing volume-wise normalization;
- New test cases to test the added classes;
- Fixed some tab/spacing mix;

For more information, please see the discussion in #1740.

### CODE REVIEW

File: spinalcordtoolbox/deepseg_gm/deepseg_gm.py [hunk 1]
```diff
@@ -183,8 +227,8 @@ def segment_volume(ninput_volume, model_name):
     axial_slices = np.asarray(axial_slices, dtype=np.float32)
     axial_slices = np.expand_dims(axial_slices, axis=3)
 
-    axial_slices -= metadata['mean_train']
-    axial_slices /= metadata['std_train']
+    normalization = VolumeStandardizationTransform()
+    axial_slices = normalization(axial_slices)
```
Comment (zougloub) [inline spinalcordtoolbox/deepseg_gm/deepseg_gm.py]: Use the more accurate term standardization (vs. normalization)?

File: spinalcordtoolbox/deepseg_gm/deepseg_gm.py [hunk 2]
```diff
@@ -113,6 +113,50 @@ def pad(self, image):
...
+        """
+        volume_mean = volume.mean()
+        volume_std = volume.std()
+
+        volume -= volume_mean
+        volume /= volume_std
+
+        return volume
+
+
```
Comment (zougloub) [inline spinalcordtoolbox/deepseg_gm/deepseg_gm.py]: A functor seems overkill.

**Comment by jcohenadad** [comment]
@perone let me know when you're happy with the state of this PR, and I'll launch the extensive test with `sct_pipeline` and add violin plots for master and this branch to compare.

File: spinalcordtoolbox/deepseg_gm/deepseg_gm.py (hunk 1 above)
Comment (perone) [reply spinalcordtoolbox/deepseg_gm/deepseg_gm.py]: Standardization is a type of normalization.

File: spinalcordtoolbox/deepseg_gm/deepseg_gm.py (hunk 2 above)
Comment (perone) [reply spinalcordtoolbox/deepseg_gm/deepseg_gm.py]: This is the usual pattern for transformations [in PyTorch for instance](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py#L120), even for the same purpose, so I don't think it's overkill, but if you want me to change, please tell me what you prefer so I can change it.

**Comment by perone** [comment]
@jcohenadad please go ahead, the functionality is this one. If something changes, it will be just minor cosmetic changes.

**Comment by perone** [comment]
@jcohenadad I'll take a look at the logs, thanks for the testing.

**Pull request merged**